### ISSUE TITLE
[🐛 Bug]: Selenium Manager downloads x86_64 firefox binaries on ARM Linux

### ISSUE THREAD

**Issue by lprimak** [SeleniumHQ/selenium#13793]
### What happened?

Firefox is available for Linux ARM (mac, OCI/ampere, ARM/gravitron etc)
However, Selenium Manager downloads the x86_64 version leading to failures

### High Impact
This issue makes impossible to run Selenium tests on Ubuntu-based docker images without going through gymnastics of installing firefox manually.
This is difficult because it requires snap, which doesn't work in Docker containers

### How can we reproduce the issue?

```shell
Any Selenium run on arm64 linux with firefox driver will reproduce this.
```

### Relevant log output

```shell
N/A
```

### Operating System

ARM Linux (Mac, cloud)

### Selenium version

Java 4.19.1 (ltest)

### What are the browser(s) and version(s) where you see this issue?

Firefox (latest)

### What are the browser driver(s) and version(s) where you see this issue?

geckodriver (latest)

### Are you using Selenium Grid?

N/A

**Comment by lprimak** [comment]
Did a bit more research, and I don't see a "download" for Arm64 Linux on Firefox' site at all. Perhaps they don't offer supported versions.
Still, I think Selenium should give an error / log or something instead of silently downloading x86 version.

Alpine linux offers firefox in their `apk` package manager, but ubuntu, debian, etc. do not.

**Comment by diemol** [comment]
@bonigarcia do you know if SM checks this?

**Comment by bonigarcia** [comment]
Indeed, Firefox is not distributed for ARM Linux. On the other hand, Selenium Manager for Linux is only compiled and distributed for x64. Maybe Selenium Manager should fail in this scenario, but I never tested it.

**Comment by diemol** [comment]
True. ARM Linux is still "rare," so we do not have the tools to test it yet. I will leave this open and see if someone wants to help add this validation. Right now, this is a low priority.

**Comment by lprimak** [comment]
I did some more research, and Mozilla is working on new distributions for Arm Linux.
See https://bugzilla.mozilla.org/show_bug.cgi?id=1678342

I didn't realize that Selenium Manager is only compiled for x64, I guess it was running in the emulation mode in my tests.

**Comment by mvysny** [comment]
Playwright supports both Chromium and Firefox for Linux arm64.

**Comment by diemol** [comment]
> Playwright supports both Chromium and Firefox for Linux arm64.

@mvysny Playwright patches Firefox, so you are not using the real browser your users have.

**Comment by atmnk** [comment]
> Indeed, Firefox is not distributed for ARM Linux. On the other hand, Selenium Manager for Linux is only compiled and distributed for x64. Maybe Selenium Manager should fail in this scenario, but I never tested it.

I see its available with focal package here https://packages.ubuntu.com/focal/arm64/firefox/download. Or I am mistaking anything. Is this still issue? Will be happy to contribute.

**Comment by diemol** [comment]
What's the Mozilla url to download Firefox for Linux on ARM architecture? We don't grab packages from Ubuntu, we grab the binary directly from Mozilla.

**Comment by atmnk** [comment]
I think they are not yet publishing for release versions they recently started publishing nightly as i can see here https://connect.mozilla.org/t5/discussions/arm64-armhf-firefox-binary-tarballs-for-packaging-for-ubuntu/m-p/12847.
May be we should wait for them to publish.

**Comment by titusfortner** [comment]
my vote is that until we can support it, we should throw an error rather than get the wrong thing

**Comment by bonigarcia** [comment]
Closed via #14099 and #14616.
